app/controllers/public_health_controller.rb

```diff
@@ -79,4 +80,14 @@ def authenticate_user_role
     # Restrict access to public health only
     redirect_to(root_url) && return unless current_user.can_view_public_health_dashboard?
   end
+
+  def set_jurisdiction_paths
+    @possible_jurisdiction_paths = current_user.jurisdiction.subtree.pluck(:id, :path).to_h
+  end
+
+  def set_all_assigned_users
+    # Get all assigned users of current user's jurisdiction
+    patients = current_user.patients&.where(jurisdiction_id: current_user.jurisdiction.subtree_ids)
```

Comment:
Any reason for the `where(jurisdiction_id: current_user.jurisdiction.subtree_ids)`? `current_user.patients` (https://github.com/SaraAlert/SaraAlert/blob/master/app/models/user.rb#L53) just calls `viewable_patients` (https://github.com/SaraAlert/SaraAlert/blob/master/app/models/user.rb#L44) which does the "scope by user jurisdiction". Removing that I'm seeing a somewhat simplified set of DB queries:

```
current_user.patients.where.not(assigned_user: nil).distinct.pluck(:assigned_user).sort

SELECT `jurisdictions`.`id` FROM `jurisdictions` WHERE ((`jurisdictions`.`ancestry` LIKE '1/2/%' OR `jurisdictions`.`ancestry` = '1/2') OR `jurisdictions`.`id` = 2)

SELECT DISTINCT `patients`.`assigned_user` FROM `patients` WHERE `patients`.`jurisdiction_id` IN (2, 3, 4) AND `patients`.`purged` = FALSE AND `patients`.`assigned_user` IS NOT NULL


current_user.patients&.where(jurisdiction_id: current_user.jurisdiction.subtree_ids).where.not(assigned_user: nil).distinct.pluck(:assigned_user).sort

SELECT `jurisdictions`.`id` FROM `jurisdictions` WHERE ((`jurisdictions`.`ancestry` LIKE '1/2/%' OR `jurisdictions`.`ancestry` = '1/2') OR `jurisdictions`.`id` = 2)

SELECT `jurisdictions`.`id` FROM `jurisdictions` WHERE ((`jurisdictions`.`ancestry` LIKE '1/2/%' OR `jurisdictions`.`ancestry` = '1/2') OR `jurisdictions`.`id` = 2)

SELECT DISTINCT `patients`.`assigned_user` FROM `patients` WHERE `patients`.`jurisdiction_id` IN (2, 3, 4) AND `patients`.`purged` = FALSE AND `patients`.`jurisdiction_id` IN (2, 3, 4) AND `patients`.`assigned_user` IS NOT NULL
```

Reply:
You are right. `current_user.patients` does exactly that. I will remove that line. Thanks!